sympy/sets/sets.py

```diff
@@ -165,7 +340,37 @@ def _union(self, other):
 
         Used within the :class:`Union` class
         """
-        return None
+        from sympy.polys.polyerrors import PolynomialError
+        from sympy.sets.fancysets import ImageSet
+
+        def _check_imageset(self):
+            # Returns True if all the args are ImageSet
+            # otherwise false
+            if isinstance(self, ImageSet):
+                return True
+            elif isinstance(self, Union):
+                for self_img in self.args:
+                    if not isinstance(self_img, ImageSet):
+                        return False
+                return True
+            else:
+                return False
+        try:
+            if _check_imageset(self) and _check_imageset(other):
+                return self._union_simplify(other)
```

Comment:
>And what is the plausibility of this bound method for a FiniteSet object 

For `FiniteSet` union [these lines]
(https://github.com/Shekharrajak/sympy/blob/fae9f28623fb9bd81baef08cc0582aba9034a26f/sympy/sets/sets.py#L1401) is being used.